Model/SepaDirectDebitPaymentMethod.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Model;
+
+use Wirecard\Oxid\Core\Helper;
+use Wirecard\Oxid\Core\PaymentMethodHelper;
+use Wirecard\Oxid\Extend\Model\Payment;
+use Wirecard\Oxid\Model\Transaction as TransactionModel;
+
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\Transaction\Transaction;
+use Wirecard\PaymentSdk\Transaction\SepaDirectDebitTransaction;
+use Wirecard\PaymentSdk\Config\SepaConfig;
+
+/**
+ * Payment method implementation for SEPA Direct Debit
+ *
+ * @since 1.0.1
+ */
+class SepaDirectDebitPaymentMethod extends PaymentMethod
+{
+    /**
+     * @inheritdoc
+     *
+     * @since 1.0.1
+     */
+    protected static $_sName = "sepadd";
+
+    /**
+     * @inheritdoc
+     *
+     * @return Config
+     *
+     * @since 1.0.1
+     */
+    public function getConfig()
+    {
+        $oConfig = parent::getConfig($this->_oPayment);
+
+        $oPaymentMethodConfig = new SepaConfig(
+            SepaDirectDebitTransaction::NAME,
+            $this->_oPayment->oxpayments__wdoxidee_maid->value,
+            $this->_oPayment->oxpayments__wdoxidee_secret->value
+        );
+
+        $oPaymentMethodConfig->setCreditorId($this->_oPayment->oxpayments__wdoxidee_creditorid->value);
+
+        $oConfig->add($oPaymentMethodConfig);
+        return $oConfig;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return Transaction
+     *
+     * @since 1.0.1
+     */
+    public function getTransaction()
+    {
+        $oTransaction = new SepaDirectDebitTransaction();
+        $oTransaction->setIban(PaymentMethodHelper::getIban());
+        $sBic = PaymentMethodHelper::getBic();
+        if ($sBic) {
+            $oTransaction->setBic($sBic);
+        }
+        $iOrderNumber = Helper::getSessionChallenge();
```

Comment:
It is not very intuitive that `getSessionChallenge` is returning an order number, is it?